crates/rattler_conda_types/src/match_spec/mod.rs

```diff
@@ -157,6 +159,8 @@ pub struct MatchSpec {
     pub url: Option<Url>,
     /// The license of the package
     pub license: Option<String>,
+    /// The condition under which this match spec applies.
+    pub condition: Option<MatchSpecCondition>,
```

Comment:
Shall we also add this to NamelessMatchSpec?

Reply:
Yeah the main challenge is that we then have a MatchSpec in a NamelessMatchspec and MatchSpec doesn't have a `Deserialize` implementation. But I think we can annotate the Condition deserializer to deserialize / serialize the MatchSpec with ToString / FromStr.